app/controllers/lti_deployments_controller.rb

```
@@ -180,11 +180,18 @@ def check_host
   end
 
   def create_course
-    new_course = Course.create!(name: params['name'], display_name: params['display_name'], is_hidden: true)
+    new_course = Course.find_or_initialize_by(name: params['name'], display_name: params['display_name'])
+    unless new_course.new_record?
+      flash_message(:error, I18n.t('lti.course_exists'))
+      redirect_to choose_course_lti_deployment_path
+      return
+    end
+    new_course.update(is_hidden: true)
+    new_course.save!
```

`update` already saves. If you want to raise an error, use `.update!`